src/frontend/AppNavigator.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import {NavigatorScreenOptions, RootStack} from './Navigation/Stack';
+import {useIntl} from 'react-intl';
+// import {SecurityContext} from '../context/SecurityContext';
+
+import BootSplash from 'react-native-bootsplash';
+import {useDeviceInfo} from './hooks/server/deviceInfo';
+import {Loading} from './sharedComponents/Loading';
+import {createDeviceNamingScreens} from './Navigation/Stack/DeviceNamingScreens';
+import {usePrefetchLastKnownLocation} from './hooks/useLastSavedLocation';
+import {usePersistedDraftObservation} from './hooks/persistedState/usePersistedDraftObservation';
+import {usePresetsQuery} from './hooks/server/presets';
+import {initializeInviteListener} from './initializeInviteListener';
+import {ProjectInviteBottomSheet} from './sharedComponents/ProjectInviteBottomSheet';
+import {getInitialRouteName} from './utils/navigation';
+import {createDefaultScreenGroup} from './Navigation/Stack/AppScreens';
+
+// Note that this does the same things as the strange syntax found in
+// https://github.com/react-navigation/react-navigation/issues/9578#issuecomment-1022991270
+// `{createScreens()}` is equivalent to `{(() => createScreens())()}`
+// Note that screen groups should have a `key` prop, so that React knows how to update them efficiently.
+
+export const AppNavigator = ({permissionAsked}: {permissionAsked: boolean}) => {
+  const {formatMessage} = useIntl();
+  const existingObservation = usePersistedDraftObservation(
+    store => store.value,
+  );
+  const {data: presets} = usePresetsQuery();
+  const deviceInfo = useDeviceInfo();
+  usePrefetchLastKnownLocation();
+  initializeInviteListener();
+
+  if (permissionAsked && !deviceInfo.isPending) {
+    BootSplash.hide();
+  }
+
+  // the user should never actually see this because the splash screen is visible, so this is to appease typescript
+  if (deviceInfo.isLoading) {
+    return <Loading />;
+  }
```

Comment:
im realizing this loading state is not necessary at all. `isPending` actually encapsulates `isLoading` (when `isPending===true` no data has been retrieved yet). So checking for `isPEnding` being false above already implies that loading is complete. So we can just remove this check completely

Reply:
https://github.com/digidem/comapeo-mobile/pull/313/commits/f97294fde2f55bc471f2a5538c5569e0b1262af0

Reply:
maybe i'm misreading, but the `isPending` check above is not doing an early return though, so the query could still be loading when it renders the main content

Reply:
OHHH   Im just seeing your edit

Reply:
think removing it introduces a potentially subtle bug. the eventual query value is being used to determine the initial route name that the navigator uses. if you use it before the query is not done loading, `!!deviceInfo.data?.name` will be `false`, which may be incorrect, and stateful updates to that value may not be used by the navigator (just based on the name of the prop, not actually sure about internal implementation)

```tsx
     <RootStack.Navigator
        initialRouteName={getInitialRouteName({
          hasDeviceName: !!deviceInfo.data?.name, // <-- only the initially available value of this is used internally
          existingObservation,
          presets,
        })}
```

Reply:
`deviceInfo.isPending` and `deviceInfo.data.name` are both derived from the same state object. So when `isPending` changes `deviceInfo.data.name` changes, therefore when `isPending===false`, `deviceInfo.data.name` will have the most up to date info and render the correct thing

Reply:
Ohh I see, your not referring to react component, but the getInitialRoute name function

Reply:
Yeah, you would be correct

Reply:
yeah i only mention the concern because of its usage in `getInitialRoute()`.

> OHHH Im just seeing your edit

yeah sorry - i have a habit of making edits instead of adding more comments to prevent notification spam 😅

Reply:
So turning it into a `suspenseQuery` is not necessarily a problem. I want to avoid using `SuspenseQueries` for smaller queries. The reason is that our suspense boundary, blocks the entire screen. So for smaller queries, it is often unnecessary, as the query only effects a small part of the ui.

That being said I think it is fine in this case because we are initializing it right away and the whole ui is blocked by the splash screen anyways. And since it is being initialized right away, the cache will be populated in later uses of it, so it will never revert back to the suspense boundary.

Reply:
https://github.com/digidem/comapeo-mobile/pull/313/commits/55683c9585f25d7df3d0a9c602148a9fc9e4fe70